test_plans/math_array.asciidoc

```diff
@@ -0,0 +1,450 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for math array
+
+This is a test plan for the math array types described in SYCL 2020 Section 4.14.3.
+
+Estimated development time is four days.
+
+== Testing scope
+
+This plan does not include negative tests.
+
+=== Backend coverage
+
+All the tests described below are performed for any SYCL implementation.
+
+=== Device coverage
+
+All the tests described below are performed only on the default device that is selected on the CTS command line.
+
+=== Type coverage
+
+The interface tests and memory layout and alignment tests (Sections 2.1 and 2.3 below) are performed using each of the following types as the underlying type the `DataT` template variable of `marray`, unless stated otherwise.
+
+In regular mode:
+
+* `char`
+* `int`
+* `float`
+* `std::byte`
+* `std::int8_t`
+* `std::int32_t`
+* A user-defined struct that meets the NumericType requirements of the C++ standard.
+
+In addition, if the device has `aspect::fp64`:
+
+* `double`
+
+In addition, if the device has `aspect::fp16`:
+
+* `half`
+
+In full conformance mode:
+
+* `signed char`
+* `unsigned char`
+* `short`
+* `unsigned short`
+* `unsigned int`
```

Comment:
The full conformance mode is intended to also include the types for regular mode. I have updated
> In full conformance mode:

to
> In full conformance mode, the following types are additionally tested:

Reply:
Ah I see. Perhaps it would be clearer to have an extensive description to avoid intensive reflection and repeat in the second part what is said in the first part to avoid forgetting some types.

Reply:
Alright, I have applied your suggestion.